### PR TITLE
2.4: empty SD card startup fixes and clean-up

### PR DESCRIPTION
**Pull request by raphaelcoeffic**
And a collection of small fixes.

### CODE REVIEW

File: radio/src/gui/colorlcd/layout.cpp [hunk 1]
```diff
@@ -75,7 +75,7 @@ void loadDefaultLayout()
   auto& screenData = g_model.screenData[0];
 
   if (screen == nullptr && defaultLayout != nullptr) {
-    strcpy(screenData.layoutName, defaultLayout->getName());
+    strcpy(screenData.layoutName, defaultLayout->getId());
```
Comment (bsongis) [inline radio/src/gui/colorlcd/layout.cpp]: layoutId then!

File: radio/src/gui/colorlcd/model_select.cpp [hunk 2]
```diff
@@ -287,12 +287,7 @@ class ModelCategoryPageBody: public FormWindow {
                 });
               }
               menu->addLine(STR_CREATE_MODEL, [=]() {
-                  storageCheck(true);
-                  modelslist.setCurrentModel(modelslist.addModel(category, createModel()));
-#if defined(LUA)
-                  // chainMenu(menuModelWizard);
-#endif
-                  update(category->size() - 1);
+                  this->createModelAction();
```
Comment (bsongis) [inline radio/src/gui/colorlcd/model_select.cpp]: why this->
should be named createModel()
or all functions would be postfixed by "Action"?

File: radio/src/gui/colorlcd/model_select.cpp [hunk 3]
```diff
@@ -342,11 +337,56 @@ class ModelCategoryPageBody: public FormWindow {
...
+      if (category->empty()) {
+        setFocus();
+      }
+    }
+
+#if defined(HARDWARE_KEYS)
+    void onEvent(event_t event) override
+    {
+      if (event == EVT_KEY_BREAK(KEY_ENTER)) {
+
```
Comment (bsongis) [inline radio/src/gui/colorlcd/model_select.cpp]: empty line?

File: radio/src/gui/colorlcd/model_select.cpp [hunk 4]
```diff
@@ -342,11 +337,56 @@ class ModelCategoryPageBody: public FormWindow {
...
+    }
+
+#if defined(HARDWARE_KEYS)
+    void onEvent(event_t event) override
+    {
+      if (event == EVT_KEY_BREAK(KEY_ENTER)) {
+    
+        Menu * menu = new Menu(this);
+        menu->addLine(STR_CREATE_MODEL, [=]() {
+          this->createModelAction();
```
Comment (bsongis) [inline radio/src/gui/colorlcd/model_select.cpp]: no this

File: radio/src/gui/colorlcd/model_select.cpp [hunk 5]
```diff
@@ -342,11 +337,56 @@ class ModelCategoryPageBody: public FormWindow {
...
+      }
+    }
+#endif
+
+    void setFocus(uint8_t flag = SET_FOCUS_DEFAULT, Window * from = nullptr) override
+    {
+      if (category->empty()) {
+        // use Window::setFocus() to avoid forwarding focus to nowhere
+        // this crashes currently in libopenui
+        Window::setFocus(flag,from);
```
Comment (bsongis) [inline radio/src/gui/colorlcd/model_select.cpp]: space needed after comma

File: radio/src/opentx.cpp [hunk 6]
```diff
@@ -391,110 +389,45 @@ void defaultInputs()
   storageDirty(EE_MODEL);
 }
 
-void applyDefaultTemplate()
+void clearMixers()
```
Comment (bsongis) [inline radio/src/opentx.cpp]: clearMixes()

File: radio/src/opentx.cpp [hunk 7]
```diff
@@ -391,110 +389,45 @@ void defaultInputs()
...
 {
-  defaultInputs(); // calls storageDirty internally
+  memset(g_model.mixData, 0, sizeof(g_model.mixData));
+}
 
+void defaultMixers()
```
Comment (bsongis) [inline radio/src/opentx.cpp]: setDefaultMixes()

File: radio/src/model_init.cpp [hunk 8]
```diff
@@ -0,0 +1,144 @@
...
+
+  // enable switch warnings
+  for (int i = 0; i < NUM_SWITCHES; i++) {
+    g_model.switchWarningState |= (1 << (3*i));
+  }
+#endif
+
+  // TODO: what about switch warnings in non-color LCD radios?
+
+  setVendorSpecificModelDefaults();
```
Comment (3djc) [inline radio/src/model_init.cpp]: move in modelDefault() after applyDefaultTemplate() for easier understanding ?

Comment (raphaelcoeffic) [reply radio/src/model_init.cpp]: Why not, I have no preferences here.

File: radio/src/gui/colorlcd/model_select.cpp (hunk 2 above)
Comment (raphaelcoeffic) [reply radio/src/gui/colorlcd/model_select.cpp]: I meant to return the lambda, in fact. Thx for reminding me!

**Pull request merged**